### ISSUE TITLE
RuntimeError: module compiled against API version a but this version of numpy is 9

### ISSUE THREAD

**Issue by tsoernes** [Theano/Theano#3623]
I used numpy 1.10 and got this error: https://github.com/Theano/Theano/issues/3052 
So I downgraded to numpy 1.9.3 and now I get this error (when importing theano):
```
RuntimeError: module compiled against API version a but this version of numpy is 9
Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\theano\gof\cutils.py", line 287, in <module>
    from cutils_ext.cutils_ext import *
ImportError: numpy.core.multiarray failed to import

During handling of the above exception, another exception occurred:
...
Problem occurred during compilation with the command line below:
C:\MinGW\bin\g++.exe -shared -g -march=corei7-avx -mcx16 -msahf -mno-movbe -maes -mpclmul -mpopcnt -mno-abm -mno-lwp -mno-fma -mno-fma4 -mno-xop -mno-bmi -mno-bmi2 -mno-tbm -mavx -mno-avx2 -msse4.2 -msse4.1 -mno-lzcnt -mno-rtm -mno-hle -mrdrnd -mf16c -mfsgsbase -mno-rdseed -mno-prfchw -mno-adx -mfxsr -mxsave -mxsaveopt --param l1-cache-size=0 --param l1-cache-line-size=0 --param l2-cache-size=256 -mtune=generic -D NPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -m32 -IC:\Python34\lib\site-packages\numpy\core\include -IC:\Python34\include -o C:\Users\Torstein\AppData\Local\Theano\compiledir_Windows-8-6.2.9200-Intel64_Family_6_Model_58_Stepping_9_GenuineIntel-3.4.3-32\cutils_ext\cutils_ext.pyd C:\Users\Torstein\AppData\Local\Theano\compiledir_Windows-8-6.2.9200-Intel64_Family_6_Model_58_Stepping_9_GenuineIntel-3.4.3-32\cutils_ext\mod.cpp -LC:\Python34\libs -LC:\Python34 -lpython34
Traceback (most recent call last):
  File "C:\Python34\lib\site-packages\theano\gof\cutils.py", line 287, in <module>
    from cutils_ext.cutils_ext import *
ImportError: numpy.core.multiarray failed to import

During handling of the above exception, another exception occurred:
```
I have this software:
Python 3.4.3 x86
Windows 10 x64
Newest MinGw as of today
pip list:
```
appdirs (1.4.0)
colorama (0.3.3)
decorator (4.0.4)
nose (1.3.7)
numpy (1.9.3)
pip (7.1.2)
py (1.4.30)
pytest (2.8.2)
pytools (2015.1.6)
scipy (0.16.1)
setuptools (12.0.5)
six (1.10.0)
Theano (0.7.0)
```
I have tried reinstalling both Theano and scipy.
`python -c "import numpy.core.multiarray"` gives no output (no errors).
Any help?

**Comment by nouiz** [comment]
Clear your Theano cache.

theano-cache purge

This should fix the second error.

Fred

**Comment by tsoernes** [comment]
I run this as admin: `python C:\Python34\Scripts\theano-cache purge` and I get this: 
```
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\myuser\\AppData\\L
ocal\\Theano\\compiledir_Windows-8-6.2.9200-Intel64_Family_6_Model_58_Stepping_9
_GenuineIntel-3.4.3-32\\cutils_ext\\cutils_ext.pyd'
```

**Comment by lamblin** [comment]
You can try to remove `C:\\Users\\myuser\\AppData\\L
ocal\\Theano\\compiledir_Windows-8-6.2.9200-Intel64_Family_6_Model_58_Stepping_9
_GenuineIntel-3.4.3-32` manually, when no python process is running, then.

**Comment by youyoujialin** [comment]
It seems that you need to downgrade all libraries depends on numpy, such as scipy, matplotlib, pandas.

**Comment by ZiangYan** [comment]
It seems that you need to downgrade all python libraries which depends on numpy, such as scipy, matplotlib, pandas, etc.

**Comment by nouiz** [comment]
That is probably the right reason. So I'll close this issue as it seem answered and isn't about Theano.

thanks

**Comment by kanwar2preet** [comment]
I faces similar issue, upgrading my pip version helped and its working:

**Comment by mellertson** [comment]
I ran into the same issue tonight. It turned out to be a problem where I had multiple numpy packages installed. An older version was installed in `/usr/lib/python2.7` and the correct version was installed in `/usr/local/lib/python2.7`.

Additionally, I had `PYTHONPATH=/usr/lib/python2.7/dist-packages:/usr/local/lib/python2.7/dist-packages`. PYTHONPATH was finding the older version of numpy before the correct version, so when inside the Python interpreter, it would import the older version of numpy.

One thing which helped was opening a python session an executing the following code:

    import numpy as np 
    print np.__version__ 
    print np.__path__

That should tell you exactly which version Python is using, and where it's installed. 

To fix the issue, I changed `PYTHONPATH=/usr/local/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages`. And I also setup a virtual Python environment using the [docs here][1]. I know I should have setup a virtual environment in the first place, but I was tired and being lazy. Oh well, lesson learned!

I hope this help!


  [1]: http://python-guide-pt-br.readthedocs.io/en/latest/dev/virtualenvs/
